spec/lib/govuk_content_security_policy_spec.rb

```diff
@@ -2,9 +2,9 @@
 require "rails"
 require "govuk_app_config/govuk_content_security_policy"
 
-RSpec.describe GovukContentSecurityPolicy do
-  class DummyCspRailsApp < Rails::Application; end
+class DummyCspRailsApp < Rails::Application; end
```

Comment:
Rather than defining a global constant, this could be set-up scoped to the test with:

```
  before do
    stub_const("DummyCspRailsApp", Class.new(Rails::Application))
  end
```

Reply:
This is actually what we'd done in the first attempt, but it causes issues where the `Rails::Application` wasn't being "registered" properly and tests that relied on that were failing... I can try and investigate further but switching away from `stub_const` fixed this.

Reply:
Ah ha sorry about that, I gave it a quick spin before commenting and didn't experience a test failure. It'll be fine if we do have to do it with a real class but it'd be good to document why.

Reply:
I was able to get it to work, but I had to make an extra change (which I believe was only working by accident before). I've explained in c956a6d.

Reply:
Great stuff. I think I've worked out what the problem is. As `Rails.application` is a class variable it ends up holding on to state in between tests. I think you can avoid the fix you've put in by adding an after block of `after { Rails.application = nil }` in both of the specs that use a Rails::Application, which can reset it after execution.

Reply:
Ooh nice, that's cleaner, I've amended the commit to do that!